### ISSUE TITLE
Unexpected unmount

### ISSUE THREAD

**Issue by FlorentCoppint** [s3ql/s3ql#220]
I encounter an issue with S3QL 3.6.0 on Ubuntu 20.10.

After running some minutes/hours, s3ql closes itself, resulting in "Transport endpoint is not connected".

Here are last debug logs : 

2020-11-18 14:37:45.663 1823540:Thread-9 s3ql.backends.s3c.close: started with s3ql_data_885473
2020-11-18 14:37:45.664 1823540:Thread-9 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885473?None, qs=None
2020-11-18 14:37:45.665 1823540:Thread-9 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885473
2020-11-18 14:37:46.996 1823540:Thread-8 s3ql.backends.common.wrapped: Average retry rate: 0.15 Hz
2020-11-18 14:37:46.997 1823540:Thread-8 s3ql.backends.common.wrapped: Encountered ConnectionTimedOut (send/recv timeout exceeded), retrying ObjectW.close (attempt 2)...
2020-11-18 14:37:47.052 1823540:Thread-8 s3ql.backends.s3c.close: started with s3ql_data_885452
2020-11-18 14:37:47.053 1823540:Thread-8 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885452?None, qs=None
2020-11-18 14:37:47.054 1823540:Thread-8 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885452
2020-11-18 14:37:48.271 1823540:Thread-7 s3ql.backends.common.wrapped: Average retry rate: 0.17 Hz
2020-11-18 14:37:48.272 1823540:Thread-7 s3ql.backends.common.wrapped: Encountered ConnectionTimedOut (send/recv timeout exceeded), retrying ObjectW.close (attempt 1)...
2020-11-18 14:37:48.295 1823540:Thread-7 s3ql.backends.s3c.close: started with s3ql_data_885466
2020-11-18 14:37:48.296 1823540:Thread-7 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885466?None, qs=None
2020-11-18 14:37:48.296 1823540:Thread-7 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885466
2020-11-18 14:37:48.781 1823540:Thread-3 s3ql.backends.common.wrapped: Average retry rate: 0.18 Hz
2020-11-18 14:37:48.781 1823540:Thread-3 s3ql.backends.common.wrapped: Encountered ConnectionTimedOut (send/recv timeout exceeded), retrying ObjectW.close (attempt 2)...
2020-11-18 14:37:48.823 1823540:Thread-3 s3ql.backends.s3c.close: started with s3ql_data_885454
2020-11-18 14:37:48.823 1823540:Thread-3 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885454?None, qs=None
2020-11-18 14:37:48.824 1823540:Thread-3 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885454
2020-11-18 14:37:49.085 1823540:Thread-4 s3ql.backends.common.wrapped: Average retry rate: 0.20 Hz
2020-11-18 14:37:49.085 1823540:Thread-4 s3ql.backends.common.wrapped: Encountered ConnectionTimedOut (send/recv timeout exceeded), retrying ObjectW.close (attempt 1)...
2020-11-18 14:37:49.115 1823540:Thread-4 s3ql.backends.s3c.close: started with s3ql_data_885465
2020-11-18 14:37:49.116 1823540:Thread-4 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885465?None, qs=None
2020-11-18 14:37:49.116 1823540:Thread-4 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885465
2020-11-18 14:37:52.754 1823540:Thread-5 s3ql.backends.common.wrapped: Average retry rate: 0.22 Hz
2020-11-18 14:37:52.754 1823540:Thread-5 s3ql.backends.common.wrapped: Encountered RequestTimeoutError (RequestTimeout: Your socket connection to the server was not read from or written to within the timeout period.), retrying ObjectW.close (attempt 1)...
2020-11-18 14:37:52.777 1823540:Thread-5 s3ql.backends.s3c.close: started with s3ql_data_885470
2020-11-18 14:37:52.778 1823540:Thread-5 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885470?None, qs=None
2020-11-18 14:37:52.778 1823540:Thread-5 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885470
2020-11-18 14:37:53.327 1823540:Thread-1 s3ql.backends.common.wrapped: Average retry rate: 0.23 Hz
2020-11-18 14:37:53.328 1823540:Thread-1 s3ql.backends.common.wrapped: Encountered RequestTimeoutError (RequestTimeout: Your socket connection to the server was not read from or written to within the timeout period.), retrying ObjectW.close (attempt 1)...
2020-11-18 14:37:53.351 1823540:Thread-1 s3ql.backends.s3c.close: started with s3ql_data_885469
2020-11-18 14:37:53.351 1823540:Thread-1 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885469?None, qs=None
2020-11-18 14:37:53.352 1823540:Thread-1 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885469
2020-11-18 14:37:53.355 1823540:Thread-1 s3ql.backends.common.wrapped: Average retry rate: 0.25 Hz
2020-11-18 14:37:53.355 1823540:Thread-1 s3ql.backends.common.wrapped: Encountered SSLError ([SSL: BAD_LENGTH] bad length (_ssl.c:2473)), retrying ObjectW.close (attempt 2)...
2020-11-18 14:37:53.390 1823540:Thread-2 s3ql.backends.common.wrapped: Average retry rate: 0.27 Hz
2020-11-18 14:37:53.391 1823540:Thread-2 s3ql.backends.common.wrapped: Encountered RequestTimeoutError (RequestTimeout: Your socket connection to the server was not read from or written to within the timeout period.), retrying ObjectW.close (attempt 1)...
2020-11-18 14:37:53.403 1823540:Thread-1 s3ql.backends.s3c.close: started with s3ql_data_885469
2020-11-18 14:37:53.403 1823540:Thread-1 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885469?None, qs=None
2020-11-18 14:37:53.404 1823540:Thread-1 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885469
2020-11-18 14:37:53.417 1823540:Thread-2 s3ql.backends.s3c.close: started with s3ql_data_885471
2020-11-18 14:37:53.417 1823540:Thread-2 s3ql.backends.s3c._do_request: started with PUT /s3ql_data_885471?None, qs=None
2020-11-18 14:37:53.417 1823540:Thread-2 s3ql.backends.s3c._send_request: sending PUT /c14-1/s3ql_data_885471
2020-11-18 14:37:53.418 1823540:Thread-2 s3ql.backends.common.wrapped: Average retry rate: 0.28 Hz
2020-11-18 14:37:53.418 1823540:Thread-2 s3ql.backends.common.wrapped: Encountered SSLError ([SSL: BAD_LENGTH] bad length (_ssl.c:2473)), retrying ObjectW.close (attempt 2)...
2020-11-18 14:37:53.422 1823540:MainThread s3ql.block_cache.with_event_loop: upload of 885470 failed
NoneType: None
2020-11-18 14:37:53.461 1823540:Thread-10 s3ql.backends.common.wrapped: Average retry rate: 0.30 Hz
2020-11-18 14:37:53.461 1823540:Thread-10 s3ql.backends.common.wrapped: Encountered RequestTimeoutError (RequestTimeout: Your socket connection to the server was not read from or written to within the timeout period.), retrying ObjectW.close (attempt 1)...
2020-11-18 14:37:53.462 1823540:Thread-5 s3ql.mount.exchook: recording exception 400 Bad request syntax ('`\x9bQ\x9b\x02')

**Comment by d--j** [comment]
This looks to me like connection problems (ConnectionTimedOut, RequestTimeoutError) and then some weird SSL errors. You backend provider probably had temporary problems.

**Comment by Nikratio** [comment]
I'm closing this issue for now. Please note that this isn't meant to imply that you haven't found a valid bug or potential improvement - you most likely have and I'm grateful that you took the time to report it.

Unfortunately, this project does not currently have any active, regular contributors or developers. The current maintainer continues to apply pull requests and tries to make regular releases, but unfortunately has no capacity to do any development beyond addressing high-impact issues. Any other reports are therefore likely to languish unless they come together with a high-quality pull request.

This report unfortunately does not fall into either category, so with the current state of affairs it's unlikely that anyone is going to be able to do anything but this anytime soon, and I prefer to use the issue tracker as a tool to manage ongoing work (as opposed to a database of known/potential issues).

**Comment by intelfx** [comment]
I'm obviously not in a position to argue, but if there's no development being done, then what's the point in reserving the issue tracker for ongoing work (of which there is none)? This amounts to just closing the issue tracker.

On the other hand, if a prospective contributor comes around with the intention of putting some work into s3ql, how would they know _what_ in fact needs work?

---

On the topic of the issue itself, though, this is mostly a special case of #227 and bad error logging. One could argue that this is a temporary network failure and as such should cause a retry rather than a crash, but this is a weird exception (Bad request syntax? Garbage in request body, huh?) and it probably does not make sense for s3ql to blanket retry all 400s.

**Comment by Nikratio** [comment]
Yes, you are right, the above rationale isn't perfectly applicable. It's a standard text block that I wrote several years ago and copy & paste since then to save time.

Let me try to give a better explanation:

I believe it is frustrating for a prospective contributor to find an issue tracker with hundreds of open issues, most of which are not easily reproducible and have been reported many years ago (so the reporters have moved on and aren't able to answer follow-up questions anymore). 

Therefore, I think the issue tracker is most useful if is curated, i.e. every issue has a clear path forward and someone is available to answer questions that relate to it. Where this is not clearly and obviously the case, I close the issue. I am also erring on the side of closing, because given the number of contributors and the amount of available work, I think it is better to close one issue that could have been addressed than to keep an issue open that turns out to be sub-optimal use of valuable contributors time.
